components/post_view/post_body_additional_content/post_body_additional_content.jsx

```diff
@@ -70,17 +70,21 @@ export default class PostBodyAdditionalContent extends React.PureComponent {
         this.generateStaticEmbed = this.generateStaticEmbed.bind(this);
         this.isLinkToggleable = this.isLinkToggleable.bind(this);
         this.handleLinkLoaded = this.handleLinkLoaded.bind(this);
-
+        const embedMetadata = props.post.metadata && props.post.metadata.embeds && props.post.metadata.embeds[0];
+        const link = embedMetadata && embedMetadata.url ? embedMetadata.url : Utils.extractFirstLink(props.post.message);
         this.state = {
-            link: Utils.extractFirstLink(props.post.message),
+            link,
             linkLoadError: false,
             linkLoaded: false,
         };
     }
 
     componentDidMount() {
         // check the availability of the image rendered(if any) in the first render.
-        this.loadShortenedImageLink();
+        const embedMetadata = this.props.post.metadata && this.props.post.metadata.embeds && this.props.post.metadata.embeds[0];
+        if (!embedMetadata || (embedMetadata && !embedMetadata.url)) {
```

Comment:
I /think/ this can be simplified to:
```suggestion
        if (!embedMetadata || !embedMetadata.url) {
```

Reply:
@lieut-data i think the suggestion change would fail if `embedMetadata` does not exist.
<img width="524" alt="Screenshot 2019-03-26 at 7 52 58 PM" src="https://user-images.githubusercontent.com/4973621/55004789-d9742380-5000-11e9-9aff-545749491e5b.png">

Reply:
Oops sorry that should work 👍 Thanks.